Remove old model usage from WooCommercePastRevenues [MAILPOET-4377]

rodrigoprimo opened the pull request:
## Description

This PR removes all old models from WooCommercePastRevenues and replaces them with Doctrine code.

## Code review notes

_N/A_

## QA notes

I'm not sure if there is much for QA to do in this PR. It changes code that is used for the CLI command `./do gen:data past_revenues`

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-4377]

## After-merge notes

_N/A_


[MAILPOET-4377]: https://mailpoet.atlassian.net/browse/MAILPOET-4377?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

rodrigoprimo committed:
Remove old NewsletterSegment model from WooCommercePastRevenues

[MAILPOET-4377]

rodrigoprimo committed:
Replace all ORM::rawExecute() calls with Doctrine code

[MAILPOET-4377]

rodrigoprimo committed:
Remove old ScheduledTaskSubscriber model from WooCommercePastRevenues

[MAILPOET-4377]

rodrigoprimo committed:
Remove old SendingQueue model from WooCommercePastRevenues

[MAILPOET-4377]

rodrigoprimo committed:
Remove old StatisticsNewsletter model from WooCommercePastRevenues

[MAILPOET-4377]

rodrigoprimo committed:
Remove old Subscriber model from WooCommercePastRevenues

[MAILPOET-4377]

rodrigoprimo committed:
Remove old Newsletter model from WooCommercePastRevenues

[MAILPOET-4377]

rodrigoprimo committed:
Remove old Segment model from WooCommercePastRevenues

[MAILPOET-4377]

samnajian commented on mailpoet/tests/DataGenerator/Generators/WooCommercePastRevenues.php:
```diff
@@ -268,59 +285,66 @@ private function getBatchLog($dataType, $generatedCount) {
-    $subscriber->save();
-    $segment->addSubscriber($subscriber->id);
+  private function createSubscriber(string $email, string $lastName, string $createdAtDate, SegmentEntity $segment, $status = SubscriberEntity::STATUS_SUBSCRIBED): SubscriberEntity {
+    $subscriber = new SubscriberEntity();
```
`MailPoet\Test\DataFactories\Subscriber` can be used to make the creation easier

rodrigoprimo replied:
I don't see any difference in this particular case in terms of making the creation easier, but at the same time, I'm not against it. I implemented this change in a new commit.

samnajian commented on mailpoet/tests/DataGenerator/Generators/WooCommercePastRevenues.php:
```diff
@@ -343,55 +367,57 @@ private function createProduct($name, $price, $categoryIds = [], $discount = 0)
-      'processed_at' => $sentAt,
-    ]);
-    $task->save();
+    $task = new ScheduledTaskEntity();
```
`MailPoet\Test\DataFactories\ScheduledTask` can facilitate creation

rodrigoprimo replied:
Please correct me if I'm wrong but I don't think there is a way to set `createdAt` and `processed` using the ScheduledTask data factory. We could add it, but I would suggest we leave this for another PR as the current approach should work fine.

samnajian replied:
I'm fine with leaving it to another PR, I only suggested doing it here since thought adding the missing part to `MailPoet\Test\DataFactories\ScheduledTask` are not very time intensive, and I leave this decision to you.

samnajian commented on mailpoet/tests/DataGenerator/Generators/WooCommercePastRevenues.php:
```diff
@@ -343,55 +367,57 @@ private function createProduct($name, $price, $categoryIds = [], $discount = 0)
-      'count_processed' => count($subscribersIds),
-    ]);
-    $queue->save();
+    $queue = new SendingQueueEntity();
```
`MailPoet\Test\DataFactories\SendingQueue` can facilitate here

rodrigoprimo replied:
Similar to the above, I don't think there is a way to use the SendingQueue data factory to set `countTotal` and `countProcessed`.

samnajian commented on mailpoet/tests/DataGenerator/Generators/WooCommercePastRevenues.php:
```diff
@@ -268,59 +285,66 @@ private function getBatchLog($dataType, $generatedCount) {
-    ORM::rawExecute("ALTER TABLE `" . $this->entityManager->getClassMetadata(StatisticsOpenEntity::class)->getTableName() . "` ENABLE KEYS");
-    ORM::rawExecute("ALTER TABLE `" . $this->entityManager->getClassMetadata(StatisticsClickEntity::class)->getTableName() . "` ENABLE KEYS");
-    ORM::rawExecute("SET UNIQUE_CHECKS = 1;");
+    $connection->executeStatement("ALTER TABLE `{$prefix}posts` ENABLE KEYS");
```
Despite these being old code, I'd suggest moving them to a different method with a meaningful name so that it's easier to know what these similar lines of code do.

rodrigoprimo replied:
I agree with the general sentiment here but I got stuck in choosing a name and thus I will suggest, if you agree, that we leave this for the future. The reason why I got stuck is that I'm not even sure if running `DISABLE KEYS` and `ENABLE KEYS` is needed. My understanding is that it only has any effect when MyISAM is being used and I would assume that the majority of the WP installs where this can be executed (mostly dev environments) are using InnoDB. I don't think it is worth investing more time in this in the context of this PR.

rodrigoprimo committed:
Use Subscriber data factory instead of SubscriberEntity

This change is in the context of creating a new subscriber.

Doing this following feedback during PR review.

[MAILPOET-4377]

rodrigoprimo commented:
Thanks for the review, @samnajian. I believe I address all your comments. Could you please take a second look?

rodrigoprimo commented:
Thanks for your review, @samnajian. I believe I replied to all your comments. Could you please take a second look?

samnajian reviewed (approved):
LGTM!
